api/v1alpha2/linodeobjectstoragekey_types.go

```diff
@@ -134,11 +133,19 @@ type LinodeObjectStorageKey struct {
 	Status LinodeObjectStorageKeyStatus `json:"status,omitempty"`
 }
 
-func (b *LinodeObjectStorageKey) GetConditions() clusterv1.Conditions {
+func (b *LinodeObjectStorageKey) GetConditions() []metav1.Condition {
 	return b.Status.Conditions
 }
 
-func (b *LinodeObjectStorageKey) SetConditions(conditions clusterv1.Conditions) {
+func (b *LinodeObjectStorageKey) SetConditions(conditions []metav1.Condition) {
+	b.Status.Conditions = conditions
+}
+
+func (b *LinodeObjectStorageKey) GetV1Beta2Conditions() []metav1.Condition {
+	return b.Status.Conditions
```

Comment:
nit: Can we call here `b.GetConditions()` or in GetConditions() `b.GetV1Beta2Conditions()` whichever is appropriate so that only one is doing get? Just to keep things simple and don't have to change multiple places in future. Same for SetConditions.

Reply:
Good idea!